ffi/src/owned.rs

```diff
@@ -209,26 +212,43 @@ impl<'itm> TryFromReprC<'itm> for &'itm str {
 
 impl IntoFfi for String {
     type Target = <Vec<u8> as IntoFfi>::Target;
+    type Store = <Vec<u8> as IntoFfi>::Store;
 
-    fn into_ffi(self) -> Self::Target {
-        self.into_bytes().into_ffi()
+    fn into_ffi(self, store: &mut MaybeUninit<Self::Store>) -> Self::Target {
+        self.into_bytes().into_ffi(store)
     }
 }
 
 impl<'slice> IntoFfi for &'slice str {
     type Target = <&'slice [u8] as IntoFfi>::Target;
+    type Store = <&'slice [u8] as IntoFfi>::Store;
 
-    fn into_ffi(self) -> Self::Target {
-        self.as_bytes().into_ffi()
+    fn into_ffi(self, store: &mut MaybeUninit<Self::Store>) -> Self::Target {
+        self.as_bytes().into_ffi(store)
     }
 }
 
 // NOTE: Arrays must be converted into pointers to be repr(C)
-unsafe impl<T: ReprC, const N: usize> ReprC for [T; N] {}
-impl<T: IntoFfi, const N: usize> IntoFfi for [T; N] {
-    type Target = LocalSlice<T::Target>;
-
-    fn into_ffi(self) -> Self::Target {
-        self.into_iter().map(IntoFfi::into_ffi).collect()
+impl<'itm, T: IntoFfi, const N: usize> IntoFfi for [T; N] {
+    type Target = *mut T::Target;
+    type Store = ([T::Target; N], [MaybeUninit<T::Store>; N]);
+
+    fn into_ffi(self, store: &mut MaybeUninit<Self::Store>) -> Self::Target {
+        let mut data_store: [MaybeUninit<_>; N] = unsafe { MaybeUninit::uninit().assume_init() };
```

Comment:
Instant UB

Reply:
no, this is literally from the [documentation](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element)

Reply:
oh, great, i was confused by: [this examples](https://doc.rust-lang.org/std/mem/union.MaybeUninit.html#initialization-invariant)